File: backend/app/schemas/hotel.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from app.schemas.common import HotelAction

class HotelModifyRequest(BaseModel):
    check_in: datetime
    check_out: Optional[datetime] = None
    action: HotelAction = HotelAction.MODIFY_CHECKIN

class HotelResponse(BaseModel):
    id: str
    itinerary_id: str
    hotel_name: str
    location: str
    check_in: datetime
    check_out: datetime
    booking_reference: Optional[str] = None
    status: str

    class Config:
        from_attributes = True

File: frontend/types/hotel.ts
export type HotelAction =
  | "NO_ACTION"
  | "MODIFY_CHECKIN"
  | "MODIFY_CHECKOUT"
  | "CANCEL"
  | "REBOOK"
  | "APPROVAL_REQUIRED";

export interface HotelBooking {
  id: string;
  itineraryId: string;
  hotelName: string;
  location: string;
  checkIn: string;
  checkOut: string;
  bookingReference: string;
  status: string;
}
